src/inmanta/server/services/environment_metrics_service.py

```diff
@@ -189,6 +192,36 @@ async def start(self) -> None:
             self.schedule(
                 self.flush_metrics, COLLECTION_INTERVAL_IN_SEC, initial_delay=COLLECTION_INTERVAL_IN_SEC, cancel_on_stop=True
             )
+            # Cleanup metrics once per hour
+            self.schedule(self._cleanup_old_metrics, interval=3600, initial_delay=0, cancel_on_stop=True)
```

Comment:
This will need to be rescheduled when the setting changes. Also, since it's an environment setting, I think it needs to be done for each environment separately. The easiest approach is probably to extract it to a method here, then allow the environment service to manage it. The periodic full compile setting does something similar. When I implemented that I did my best to do it in a generic way so new schedule settings could easily be added.

- `_enable_schedules` [enables all relevant schedules for the environment](https://github.com/inmanta/inmanta-core/blob/6e7e51da8c8683dc0d7bdb47879c9115cf570f2c/src/inmanta/server/services/environmentservice.py#L166). It is called when the environment [is created](https://github.com/inmanta/inmanta-core/blob/6e7e51da8c8683dc0d7bdb47879c9115cf570f2c/src/inmanta/server/services/environmentservice.py#L389), [starts](https://github.com/inmanta/inmanta-core/blob/6e7e51da8c8683dc0d7bdb47879c9115cf570f2c/src/inmanta/server/services/environmentservice.py#L156) or when [a setting is updated](https://github.com/inmanta/inmanta-core/blob/6e7e51da8c8683dc0d7bdb47879c9115cf570f2c/src/inmanta/server/services/environmentservice.py#L202).
- It delegates the actual scheduling to the compiler service, which [keeps track of existing schedules so it can replace them](https://github.com/inmanta/inmanta-core/blob/6e7e51da8c8683dc0d7bdb47879c9115cf570f2c/src/inmanta/server/services/compilerservice.py#L555). I can't say for sure off the top of my head but it's possible interval schedules are replaced automatically.